sdk/metric/sdk.go

```diff
@@ -391,62 +381,91 @@ func (m *SDK) Labels(kvs ...core.KeyValue) api.LabelSet {
 		oi++
 	}
 	kvs = kvs[0:oi]
-	ls.slice = kvs
 	ls.computeOrdered(kvs)
 	return ls
 }
 
+func (ls *labels) ensureCachedValue() {
```

Comment:
Making this calculation lazy makes me nervous. I suspect some exporter one day will want to use multiple goroutines and then, we might have race conditions.  If we assume that each export record is actually use, might this computation be better done eagerly?

Reply:
Yeah, will make it eager then.